guvnor-m2repo-editor/guvnor-m2repo-editor-client/src/main/java/org/guvnor/m2repo/client/widgets/ArtifactListPresenterImpl.java

```diff
@@ -36,58 +35,67 @@
 public class ArtifactListPresenterImpl
         implements ArtifactListPresenter {
 
-    @Inject
-    protected ArtifactListViewImpl view;
+    static final boolean DEFAULT_ORDER_ASCENDING = true;
+    private final ArtifactListView view;
 
-    @Inject
-    private Caller<M2RepoService> m2RepoService;
+    private final Caller<M2RepoService> m2RepoService;
 
-    @Inject
-    private Event<NotificationEvent> notification;
+    private final Event<NotificationEvent> notification;
 
+    // Initialized lazily on the first refresh
     private RefreshableAsyncDataProvider dataProvider;
 
+    @Inject
+    public ArtifactListPresenterImpl( ArtifactListView view, Caller<M2RepoService> m2RepoService, Event<NotificationEvent> notification ) {
+        this.view = view;
+        this.m2RepoService = m2RepoService;
+        this.notification = notification;
+    }
+
     @PostConstruct
     public void init() {
         view.init( this );
+        dataProvider = new RefreshableAsyncDataProvider( view, m2RepoService );
+        dataProvider.addDataDisplay( view.getDisplay() );
+    }
+
+    @Override
+    public ArtifactListView getView() {
+        return view;
     }
 
     @Override
     public void refresh() {
-        // We couldn't attach the Display to the DataProvider in the @PostConstruct method
-        // as this initializes the Display with data using AsyncDataProvider.onRangeChanged()
-        // at which time the View may not have been initialised fully.
```

Comment:
@manstis Do you remember what exactly did you mean by *the view may have not been initialised fully*? IMO the view instance is already fully initialized when injected into presenter constructor. The only thing I can imagine is that an additional column is [conditionally added](https://github.com/droolsjbpm/guvnor/blob/master/guvnor-m2repo-editor/guvnor-m2repo-editor-client/src/main/java/org/guvnor/m2repo/client/editor/MavenRepositoryPagedJarTable.java#L56-L78) in parent component's `@PostConstruct` but I have tested the change manually and everything works as expected.